File: backend/laundryghar/laundryghar.SharedDataModel/Enums/UserType.cs
namespace laundryghar.SharedDataModel.Enums;

/// <summary>
/// The coarse account/role type used for auth scope resolution. Most values are vertical-neutral;
/// <see cref="WarehouseStaff"/> is the laundry-specific operational-staff label retained for data
/// compatibility (seeded users/roles + DB CHECKs reference <c>warehouse_staff</c>). New verticals
/// should use the neutral <see cref="OpsStaff"/> for their on-site processing/service staff.
/// (Neutralized in multi-vertical Phase 2 / slice 2D.)
/// </summary>
public static class UserType
{
    public const string PlatformAdmin = "platform_admin";
    public const string BrandAdmin = "brand_admin";
    public const string FranchiseOwner = "franchise_owner";
    public const string StoreAdmin = "store_admin";
    public const string Staff = "staff";

    /// <summary>Laundry-specific operational staff (warehouse wash/QC). Retained for data
    /// compatibility; prefer <see cref="OpsStaff"/> for vertical-neutral processing staff.</summary>
    public const string WarehouseStaff = "warehouse_staff";

    /// <summary>Vertical-neutral on-site operational/processing staff (salon stylist,
    /// logistics hub operator, …) — the generic successor to <see cref="WarehouseStaff"/>.</summary>
    public const string OpsStaff = "ops_staff";

    public const string Rider = "rider";
    public const string Auditor = "auditor";
    public const string Support = "support";

    public static readonly IReadOnlySet<string> All = new HashSet<string>
    {
        PlatformAdmin, BrandAdmin, FranchiseOwner, StoreAdmin, Staff,
        WarehouseStaff, OpsStaff, Rider, Auditor, Support,
    };

    public static bool IsValid(string? value) => value is not null && All.Contains(value);

    /// <summary>True if the type is on-site operational/processing staff in any vertical
    /// (laundry <c>warehouse_staff</c> or the neutral <c>ops_staff</c>).</summary>
    public static bool IsOperationalStaff(string? value)
        => value is WarehouseStaff or OpsStaff;
}

File: admin-web/src/types/userType.ts
/**
 * The coarse account user_type vocabulary, mirroring the backend
 * laundryghar.SharedDataModel/Enums/UserType. Keep in sync with that file.
 *
 * `warehouse_staff` is the laundry-specific on-site staff type retained for data compatibility;
 * `ops_staff` is the vertical-neutral successor used by salon/logistics. See verticalTerms for which
 * one a given brand should use.
 */
export const UserType = {
  platformAdmin: 'platform_admin',
  brandAdmin: 'brand_admin',
  franchiseOwner: 'franchise_owner',
  storeAdmin: 'store_admin',
  staff: 'staff',
  warehouseStaff: 'warehouse_staff',
  opsStaff: 'ops_staff',
  rider: 'rider',
  auditor: 'auditor',
  support: 'support',
} as const

export type UserTypeKey = (typeof UserType)[keyof typeof UserType]

/** Human-readable labels for each user_type — vertical-neutral. */
export const USER_TYPE_LABEL: Record<string, string> = {
  platform_admin: 'Platform admin',
  brand_admin: 'Brand admin',
  franchise_owner: 'Franchise owner',
  store_admin: 'Store admin',
  staff: 'Staff',
  warehouse_staff: 'Warehouse staff',
  ops_staff: 'Operations staff',
  rider: 'Rider',
  auditor: 'Auditor',
  support: 'Support',
}

/** Humanise a user_type code, falling back to the raw code if unknown. */
export function userTypeLabel(userType: string | null | undefined): string {
  if (!userType) return '—'
  return USER_TYPE_LABEL[userType] ?? userType
}
